lib/api/controllers/routerController.js

```diff
@@ -226,31 +226,33 @@ module.exports = function RouterController (kuzzle) {
     });
 
     apiBase.get('/swagger.json', (request, response) => {
-      var content;
-      try {
-        content = fs.readFileSync(__dirname + '/../../../kuzzle-swagger.json');
-      }
-      catch (e) {
-        content = stringify({error: 'The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it.', status: 500});
-      }
       response.setHeader('Access-Control-Allow-Origin', '*');
       response.setHeader('Access-Control-Allow-Headers', 'X-Requested-With');
-      response.writeHead(200, {'Content-Type': 'application/json'});
-      response.end(content);
+
+      fs.readFile(__dirname + '/../../../kuzzle-swagger.json', (err, data) => {
+        if (err) {
+          response.writeHead(500, {'Content-Type': 'application/json'});
+          reponse.end(JSON.stringify({error: 'The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it.', status: 500}))
+        }
+
+        response.writeHead(200, {'Content-Type': 'application/json'});
+        response.end(data);
+      });
     });
 
     apiBase.get('/swagger.yml', (request, response) => {
-      var content;
-      try {
-        content = fs.readFileSync(__dirname + '/../../../kuzzle-swagger.yml');
-      }
-      catch (e) {
-        content = '  error: "The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it."\n  status: 500';
-      }
       response.setHeader('Access-Control-Allow-Origin', '*');
       response.setHeader('Access-Control-Allow-Headers', 'X-Requested-With');
-      response.writeHead(200, {'Content-Type': 'application/yaml'});
-      response.end(content);
+
+      fs.readFile(__dirname + '/../../../kuzzle-swagger.yml', (err, data) => {
+        if (err) {
+          response.writeHead(500, {'Content-Type': 'application/yaml'});
+          reponse.end(JSON.stringify({error: 'The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it.', status: 500}))
```

Comment:
you need to return from the function after sending the error response, otherwise you'll send the data one too (with undefined data might I add)